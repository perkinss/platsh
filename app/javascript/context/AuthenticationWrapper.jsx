import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
