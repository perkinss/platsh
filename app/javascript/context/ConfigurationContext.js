import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
