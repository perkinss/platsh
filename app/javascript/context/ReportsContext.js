import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
