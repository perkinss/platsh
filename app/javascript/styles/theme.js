import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
