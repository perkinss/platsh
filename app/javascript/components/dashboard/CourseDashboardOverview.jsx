import React, { useContext, useEffect } from 'react'
import { CircularProgress, Paper } from '@material-ui/core'
import CourseDashboard from './CourseDashboard'
import {ConfigurationContext} from "../../context/ConfigurationContext";
import {withStyles} from "@material-ui/core/styles/index";

const styles = theme => ({
    root: {
        flexGrow: 1,
    },
    paper: {
        padding: theme.spacing(2),
        marginTop: theme.spacing(3),
        marginBottom: theme.spacing(3),
        textAlign: 'center',
        color: theme.palette.text.secondary,
    },
    toolbar: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'flex-end',
        padding: '0 8px',
        ...theme.mixins.toolbar,
    },
    content: {
        flexGrow: 1,
        padding: theme.spacing(1),
    },
    section: {
        padding: theme.spacing(.5),
        margin: theme.spacing(3, 0, 5, 0)
    }
})


function CourseDashboardOverview(props) {
    const { classes } = props
    const { state } = useContext(ConfigurationContext)

    const courses = state.courses

    return (
        <div className={classes.content}>
            {!courses && <Paper className={classes.paper}> <CircularProgress className={classes.progress} color={'secondary'} /></Paper>}

        {courses && Object.keys(courses).length > 0 &&
            <React.Fragment>

                {Array.from(Object.keys(courses)).map(courseId => {
                    return (
                        <CourseDashboard courseId={courseId} course={courses[courseId]} classes={classes} key={`course-${courses[courseId].title}`}/>
                    )
                })}
            </React.Fragment>

        }
        </div>

    )
}
export default withStyles(styles)(CourseDashboardOverview);
