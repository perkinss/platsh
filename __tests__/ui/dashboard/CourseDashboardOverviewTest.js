import React from 'react'
import { createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import CourseDashboardOverview from "../../../app/javascript/components/dashboard/CourseDashboardOverview";
import { ConfigurationContext } from "../../../app/javascript/context/ConfigurationContext";
import theme from "../../../app/javascript/styles/theme";
import { ThemeProvider } from '@material-ui/styles'

Enzyme.configure({ adapter: new Adapter() });

describe('<CourseDashboardOverview />', () => {
    let render;
    let classes = {

    }
    let state = {
        courses: {
            1: {
                title: "Math 9",
                sections: {
                    17:
                    {
                        name: "New at the Zoo...",
                    }
                }
            }}}
    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <CourseDashboardOverview classes={classes} />
                </ConfigurationContext.Provider>
            </ThemeProvider>
                );
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <CourseDashboardOverview classes={classes} />
                </ConfigurationContext.Provider>
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toContain("Math 9")
        expect(result).toContain("CourseDashboard-expand")
        // everything hidden by default.

    });



})