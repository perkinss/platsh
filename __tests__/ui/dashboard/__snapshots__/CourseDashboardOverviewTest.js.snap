// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseDashboardOverview /> should match the snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "CourseDashboardOverview-content-4",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiPaper-root CourseDashboard-paper-7 CourseDashboardOverview-paper-2 MuiPaper-elevation1 MuiPaper-rounded",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft",
            },
            "children": Array [
              Node {
                "data": "Math 9",
                "next": Node {
                  "attribs": Object {
                    "aria-expanded": "false",
                    "aria-label": "show more",
                    "class": "MuiButtonBase-root MuiIconButton-root CourseDashboard-expand-11",
                    "tabindex": "0",
                    "type": "button",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "class": "MuiIconButton-label",
                      },
                      "children": Array [
                        Node {
                          "attribs": Object {
                            "aria-hidden": "true",
                            "class": "MuiSvgIcon-root",
                            "focusable": "false",
                            "viewBox": "0 0 24 24",
                          },
                          "children": Array [
                            Node {
                              "attribs": Object {
                                "d": "M15.88 9.29L12 13.17 8.12 9.29a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41-.39-.38-1.03-.39-1.42 0z",
                              },
                              "children": Array [],
                              "name": "path",
                              "namespace": "http://www.w3.org/2000/svg",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "d": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "d": undefined,
                              },
                            },
                          ],
                          "name": "svg",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "aria-hidden": undefined,
                            "class": undefined,
                            "focusable": undefined,
                            "viewBox": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "aria-hidden": undefined,
                            "class": undefined,
                            "focusable": undefined,
                            "viewBox": undefined,
                          },
                        },
                      ],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-expanded": undefined,
                    "aria-label": undefined,
                    "class": undefined,
                    "tabindex": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-expanded": undefined,
                    "aria-label": undefined,
                    "class": undefined,
                    "tabindex": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              Node {
                "attribs": Object {
                  "aria-expanded": "false",
                  "aria-label": "show more",
                  "class": "MuiButtonBase-root MuiIconButton-root CourseDashboard-expand-11",
                  "tabindex": "0",
                  "type": "button",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "class": "MuiIconButton-label",
                    },
                    "children": Array [
                      Node {
                        "attribs": Object {
                          "aria-hidden": "true",
                          "class": "MuiSvgIcon-root",
                          "focusable": "false",
                          "viewBox": "0 0 24 24",
                        },
                        "children": Array [
                          Node {
                            "attribs": Object {
                              "d": "M15.88 9.29L12 13.17 8.12 9.29a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41-.39-.38-1.03-.39-1.42 0z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                        ],
                        "name": "svg",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "viewBox": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "viewBox": undefined,
                        },
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "data": "Math 9",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-expanded": undefined,
                  "aria-label": undefined,
                  "class": undefined,
                  "tabindex": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-expanded": undefined,
                  "aria-label": undefined,
                  "class": undefined,
                  "tabindex": undefined,
                  "type": undefined,
                },
              },
            ],
            "name": "h4",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
